src/test/java/org/apache/cassandra/sidecar/MainModuleTest.java

```diff
@@ -0,0 +1,25 @@
+package org.apache.cassandra.sidecar;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.name.Names;
+import com.google.inject.util.Modules;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Unit tests for the {@link MainModule} class
+ */
+public class MainModuleTest {
+    @Test
+    public void testSidecarVersion()
+    {
+        Injector injector = Guice.createInjector(Modules.override(new MainModule()).with(new TestModule()));
+
+        String sidecarVersion = injector.getInstance(Key.get(String.class, Names.named("SidecarVersion")));
+
+        assertEquals("1.0-TEST", sidecarVersion);
```

Comment:
Thanks for adding the test, but it does not run successfully. 
This is the result I get
```
org.opentest4j.AssertionFailedError: expected: <1.0-TEST> but was: <1.0-SNAPSHOT>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1142)
	at org.apache.cassandra.sidecar.MainModuleTest.testSidecarVersion(MainModuleTest.java:23)
```

Reply:
What happens is that `:jar` task runs before `:test` task. So the test actually reads from the snapshot jar built.

Reply:
Added the missing test resource file.